<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns='http://www.w3.org/1999/xhtml' xml:lang='en' lang='en'>
<head>
<title>buildroot - Buildroot: Making Embedded Linux easy</title>
<meta name='generator' content='cgit v0.9.1'/>
<meta name='robots' content='index, nofollow'/>
<link rel='stylesheet' type='text/css' href='/cgit.css'/>
<link rel='alternate' title='Atom feed' href='http://git.buildroot.net/buildroot/atom/package/lua/5.1.5/0012-fix-reader-at-eoz.patch?h=master' type='application/atom+xml'/>
</head>
<body>
<div id='cgit'><table id='header'>
<tr>
<td class='logo' rowspan='2'><a href='/'><img src='/busybox1.png' alt='cgit logo'/></a></td>
<td class='main'><a href='/'>index</a> : <a title='buildroot' href='/buildroot/'>buildroot</a></td><td class='form'><form method='get' action=''>
<select name='h' onchange='this.form.submit();'>
<option value='2012.11.x'>2012.11.x</option>
<option value='2013.08.x'>2013.08.x</option>
<option value='master' selected='selected'>master</option>
<option value='next'>next</option>
</select> <input type='submit' name='' value='switch'/></form></td></tr>
<tr><td class='sub'>Buildroot: Making Embedded Linux easy</td><td class='sub right'>jacmet</td></tr></table>
<table class='tabs'><tr><td>
<a href='/buildroot/'>summary</a><a href='/buildroot/refs/'>refs</a><a href='/buildroot/log/package/lua/5.1.5/0012-fix-reader-at-eoz.patch'>log</a><a class='active' href='/buildroot/tree/package/lua/5.1.5/0012-fix-reader-at-eoz.patch'>tree</a><a href='/buildroot/commit/package/lua/5.1.5/0012-fix-reader-at-eoz.patch'>commit</a><a href='/buildroot/diff/package/lua/5.1.5/0012-fix-reader-at-eoz.patch'>diff</a></td><td class='form'><form class='right' method='get' action='/buildroot/log/package/lua/5.1.5/0012-fix-reader-at-eoz.patch'>
<select name='qt'>
<option value='grep'>log msg</option>
<option value='author'>author</option>
<option value='committer'>committer</option>
<option value='range'>range</option>
</select>
<input class='txt' type='text' size='10' name='q' value=''/>
<input type='submit' value='search'/>
</form>
</td></tr></table>
<div class='path'>path: <a href='/buildroot/tree/'>root</a>/<a href='/buildroot/tree/package'>package</a>/<a href='/buildroot/tree/package/lua'>lua</a>/<a href='/buildroot/tree/package/lua/5.1.5'>5.1.5</a>/<a href='/buildroot/tree/package/lua/5.1.5/0012-fix-reader-at-eoz.patch'>0012-fix-reader-at-eoz.patch</a></div><div class='content'>blob: 17a28ba4c337cc5742d341acb54b53f3a5d94909 (<a href='/buildroot/plain/package/lua/5.1.5/0012-fix-reader-at-eoz.patch'>plain</a>)
<table summary='blob content' class='blob'>
<tr><td class='linenumbers'><pre><a class='no' id='n1' name='n1' href='#n1'>1</a>
<a class='no' id='n2' name='n2' href='#n2'>2</a>
<a class='no' id='n3' name='n3' href='#n3'>3</a>
<a class='no' id='n4' name='n4' href='#n4'>4</a>
<a class='no' id='n5' name='n5' href='#n5'>5</a>
<a class='no' id='n6' name='n6' href='#n6'>6</a>
<a class='no' id='n7' name='n7' href='#n7'>7</a>
<a class='no' id='n8' name='n8' href='#n8'>8</a>
<a class='no' id='n9' name='n9' href='#n9'>9</a>
<a class='no' id='n10' name='n10' href='#n10'>10</a>
<a class='no' id='n11' name='n11' href='#n11'>11</a>
<a class='no' id='n12' name='n12' href='#n12'>12</a>
<a class='no' id='n13' name='n13' href='#n13'>13</a>
<a class='no' id='n14' name='n14' href='#n14'>14</a>
<a class='no' id='n15' name='n15' href='#n15'>15</a>
<a class='no' id='n16' name='n16' href='#n16'>16</a>
<a class='no' id='n17' name='n17' href='#n17'>17</a>
<a class='no' id='n18' name='n18' href='#n18'>18</a>
<a class='no' id='n19' name='n19' href='#n19'>19</a>
<a class='no' id='n20' name='n20' href='#n20'>20</a>
<a class='no' id='n21' name='n21' href='#n21'>21</a>
<a class='no' id='n22' name='n22' href='#n22'>22</a>
<a class='no' id='n23' name='n23' href='#n23'>23</a>
<a class='no' id='n24' name='n24' href='#n24'>24</a>
<a class='no' id='n25' name='n25' href='#n25'>25</a>
<a class='no' id='n26' name='n26' href='#n26'>26</a>
<a class='no' id='n27' name='n27' href='#n27'>27</a>
<a class='no' id='n28' name='n28' href='#n28'>28</a>
<a class='no' id='n29' name='n29' href='#n29'>29</a>
<a class='no' id='n30' name='n30' href='#n30'>30</a>
<a class='no' id='n31' name='n31' href='#n31'>31</a>
<a class='no' id='n32' name='n32' href='#n32'>32</a>
<a class='no' id='n33' name='n33' href='#n33'>33</a>
<a class='no' id='n34' name='n34' href='#n34'>34</a>
<a class='no' id='n35' name='n35' href='#n35'>35</a>
<a class='no' id='n36' name='n36' href='#n36'>36</a>
<a class='no' id='n37' name='n37' href='#n37'>37</a>
<a class='no' id='n38' name='n38' href='#n38'>38</a>
<a class='no' id='n39' name='n39' href='#n39'>39</a>
<a class='no' id='n40' name='n40' href='#n40'>40</a>
<a class='no' id='n41' name='n41' href='#n41'>41</a>
<a class='no' id='n42' name='n42' href='#n42'>42</a>
<a class='no' id='n43' name='n43' href='#n43'>43</a>
<a class='no' id='n44' name='n44' href='#n44'>44</a>
<a class='no' id='n45' name='n45' href='#n45'>45</a>
<a class='no' id='n46' name='n46' href='#n46'>46</a>
</pre></td>
<td class='lines'><pre><code>When loading a file, Lua may call the reader function again after it returned end of input.

Fetch from: http://www.lua.org/bugs.html#5.1.5-2

Signed-off-by: Francois Perrad &lt;francois.perrad@gadz.org&gt;

Index: b/src/lzio.c
===================================================================
--- a/src/lzio.c
+++ b/src/lzio.c
@@ -22,10 +22,14 @@
   size_t size;
   lua_State *L = z-&gt;L;
   const char *buff;
+  if (z-&gt;eoz) return EOZ;
   lua_unlock(L);
   buff = z-&gt;reader(L, z-&gt;data, &amp;size);
   lua_lock(L);
-  if (buff == NULL || size == 0) return EOZ;
+  if (buff == NULL || size == 0) {
+    z-&gt;eoz = 1;  /* avoid calling reader function next time */
+    return EOZ;
+  }
   z-&gt;n = size - 1;
   z-&gt;p = buff;
   return char2int(*(z-&gt;p++));
@@ -51,6 +55,7 @@
   z-&gt;data = data;
   z-&gt;n = 0;
   z-&gt;p = NULL;
+  z-&gt;eoz = 0;
 }
 
 
Index: b/src/lzio.h
===================================================================
--- a/src/lzio.h
+++ b/src/lzio.h
@@ -59,6 +59,7 @@
   lua_Reader reader;
   void* data;			/* additional data */
   lua_State *L;			/* Lua state (for reader) */
+  int eoz;			/* true if reader has no more data */
 };
 
 
</code></pre></td></tr></table>
</div> <!-- class=content -->
<div class='footer'>generated  by cgit v0.9.1 at 2015-02-18 09:43:24 (GMT)</div>
</div> <!-- id=cgit -->
</body>
</html>
